internal/common/auth/authorization/kubernetes.go

```diff
@@ -161,12 +183,20 @@ func (authService *KubernetesNativeAuthService) getClusterURL(token string) (str
 func (authService *KubernetesNativeAuthService) reviewToken(ctx context.Context, clusterUrl string, token string, ca []byte) (string, error) {
 	result, err := authService.TokenReviewer.ReviewToken(ctx, clusterUrl, token, ca)
 	if err != nil {
-		return "", err
+		// TODO(clif) Hard to tell if this should be internal auth error
+		// or invalid creds still.
```

Comment:
Does ReviewToken() return different errors that delineate if it's a mangled/bad token as opposed to the auth server erroring?

Reply:
Really hard to tell, as the call just seems to end up doing a `POST` to this kubernetes endpoint: https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-review-v1/ Doesn't seem to delineate what errors could be returned.